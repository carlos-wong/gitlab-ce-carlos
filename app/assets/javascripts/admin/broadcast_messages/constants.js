import { s__ } from '~/locale';

export const TYPE_BANNER = 'banner';
export const TYPE_NOTIFICATION = 'notification';

export const TYPES = [
  { value: TYPE_BANNER, text: s__('BroadcastMessages|Banner') },
  { value: TYPE_NOTIFICATION, text: s__('BroadcastMessages|Notification') },
];

export const THEMES = [
  { value: 'indigo', text: s__('BroadcastMessages|Indigo') },
  { value: 'light-indigo', text: s__('BroadcastMessages|Light Indigo') },
  { value: 'blue', text: s__('BroadcastMessages|Blue') },
  { value: 'light-blue', text: s__('BroadcastMessages|Light Blue') },
  { value: 'green', text: s__('BroadcastMessages|Green') },
  { value: 'light-green', text: s__('BroadcastMessages|Light Green') },
  { value: 'red', text: s__('BroadcastMessages|Red') },
  { value: 'light-red', text: s__('BroadcastMessages|Light Red') },
  { value: 'dark', text: s__('BroadcastMessages|Dark') },
  { value: 'light', text: s__('BroadcastMessages|Light') },
];

export const NEW_BROADCAST_MESSAGE = {
  message: '',
  broadcastType: TYPES[0].value,
  theme: THEMES[0].value,
  dismissable: false,
  targetPath: '',
  targetAccessLevels: [],
  startsAt: new Date(),
  endsAt: new Date(),
};
