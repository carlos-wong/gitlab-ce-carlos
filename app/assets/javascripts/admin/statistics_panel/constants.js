import { s__ } from '~/locale';

const statisticsLabels = {
  forks: s__('AdminStatistics|Forks'),
  issues: s__('AdminStatistics|Issues'),
  mergeRequests: s__('AdminStatistics|Merge Requests'),
  notes: s__('AdminStatistics|Notes'),
  snippets: s__('AdminStatistics|Snippets'),
  sshKeys: s__('AdminStatistics|SSH Keys'),
  milestones: s__('AdminStatistics|Milestones'),
  activeUsers: s__('AdminStatistics|Active Users'),
};

export default statisticsLabels;
