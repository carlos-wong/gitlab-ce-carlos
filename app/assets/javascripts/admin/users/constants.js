import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};
