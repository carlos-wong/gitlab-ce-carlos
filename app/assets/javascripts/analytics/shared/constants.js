import { masks } from 'dateformat';
import { s__ } from '~/locale';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};

// Some content is duplicated due to backward compatibility.
// It will be removed with https://gitlab.com/gitlab-org/gitlab/-/issues/350614 in 14.9
export const METRICS_POPOVER_CONTENT = {
  lead_time: {
    description: s__('ValueStreamAnalytics|Median time from issue created to issue closed.'),
  },
  cycle_time: {
    description: s__(
      "ValueStreamAnalytics|Median time from the earliest commit of a linked issue's merge request to when that issue is closed.",
    ),
  },
  lead_time_for_changes: {
    description: s__(
      'ValueStreamAnalytics|Median time between merge request merge and deployment to a production environment for all MRs deployed in the given time period.',
    ),
  },
  issues: { description: s__('ValueStreamAnalytics|Number of new issues created.') },
  deploys: { description: s__('ValueStreamAnalytics|Total number of deploys to production.') },
  deployment_frequency: {
    description: s__('ValueStreamAnalytics|Average number of deployments to production per day.'),
  },
  commits: {
    description: s__('ValueStreamAnalytics|Number of commits pushed to the default branch'),
  },
  time_to_restore_service: {
    description: s__(
      'ValueStreamAnalytics|Median time an incident was open on a production environment in the given time period.',
    ),
  },
  change_failure_rate: {
    description: s__(
      'ValueStreamAnalytics|Percentage of deployments that cause an incident in production.',
    ),
  },
};

const KEY_METRICS_TITLE = s__('ValueStreamAnalytics|Key metrics');
const KEY_METRICS_KEYS = ['lead_time', 'cycle_time', 'issues', 'commits', 'deploys'];

const DORA_METRICS_TITLE = s__('ValueStreamAnalytics|DORA metrics');
const DORA_METRICS_KEYS = [
  'deployment_frequency',
  'lead_time_for_changes',
  'time_to_restore_service',
  'change_failure_rate',
];

export const VSA_METRICS_GROUPS = [
  { key: 'key_metrics', title: KEY_METRICS_TITLE, keys: KEY_METRICS_KEYS },
  { key: 'dora_metrics', title: DORA_METRICS_TITLE, keys: DORA_METRICS_KEYS },
];
