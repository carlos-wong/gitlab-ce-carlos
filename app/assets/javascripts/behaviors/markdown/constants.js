// https://prosemirror.net/docs/ref/#model.ParseRule.priority
export const DEFAULT_PARSE_RULE_PRIORITY = 50;
export const HIGHER_PARSE_RULE_PRIORITY = 1 + DEFAULT_PARSE_RULE_PRIORITY;
