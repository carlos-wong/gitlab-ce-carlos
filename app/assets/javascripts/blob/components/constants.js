import { __ } from '~/locale';

export const BTN_COPY_CONTENTS_TITLE = __('Copy file contents');
export const BTN_RAW_TITLE = __('Open raw');
export const BTN_DOWNLOAD_TITLE = __('Download');

export const SIMPLE_BLOB_VIEWER = 'simple';
export const SIMPLE_BLOB_VIEWER_TITLE = __('Display source');

export const RICH_BLOB_VIEWER = 'rich';
export const RICH_BLOB_VIEWER_TITLE = __('Display rendered file');
