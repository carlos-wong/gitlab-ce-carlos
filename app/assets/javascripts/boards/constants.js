export const ListType = {
  assignee: 'assignee',
  milestone: 'milestone',
  backlog: 'backlog',
  closed: 'closed',
  label: 'label',
};

export default {
  ListType,
};
