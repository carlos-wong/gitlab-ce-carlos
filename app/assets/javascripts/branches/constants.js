export const SIDES = {
  full: 'full',
  left: 'left',
  right: 'right',
};
export const MAX_COMMIT_COUNT = 1000;
