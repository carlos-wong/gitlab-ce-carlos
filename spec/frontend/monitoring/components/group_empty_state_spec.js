import { shallowMount } from '@vue/test-utils';
import GroupEmptyState from '~/monitoring/components/group_empty_state.vue';
import { metricStates } from '~/monitoring/constants';

function createComponent(props) {
  return shallowMount(GroupEmptyState, {
    propsData: {
      ...props,
      documentationPath: '/path/to/docs',
      settingsPath: '/path/to/settings',
      svgPath: '/path/to/empty-group-illustration.svg',
    },
  });
}

describe('GroupEmptyState', () => {
  const supportedStates = [
    metricStates.NO_DATA,
    metricStates.TIMEOUT,
    metricStates.CONNECTION_FAILED,
    metricStates.BAD_QUERY,
    metricStates.LOADING,
    metricStates.UNKNOWN_ERROR,
    'FOO STATE', // does not fail with unknown states
  ];

  test.each(supportedStates)('Renders an empty state for %s', selectedState => {
    const wrapper = createComponent({ selectedState });

    expect(wrapper.element).toMatchSnapshot();
    // slot is not rendered by the stub, test it separately
    expect(wrapper.vm.currentState.slottedDescription).toMatchSnapshot();
  });
});
