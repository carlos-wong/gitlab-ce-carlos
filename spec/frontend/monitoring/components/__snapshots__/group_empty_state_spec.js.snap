// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupEmptyState Renders an empty state for BAD_QUERY 1`] = `
<gl-empty-state-stub
  compact="true"
  primarybuttonlink="/path/to/settings"
  primarybuttontext="Verify configuration"
  svgpath="/path/to/empty-group-illustration.svg"
  title="Query cannot be processed"
/>
`;

exports[`GroupEmptyState Renders an empty state for BAD_QUERY 2`] = `"The Prometheus server responded with \\"bad request\\". Please check your queries are correct and are supported in your Prometheus version. <a href=\\"/path/to/docs\\">More information</a>"`;

exports[`GroupEmptyState Renders an empty state for CONNECTION_FAILED 1`] = `
<gl-empty-state-stub
  compact="true"
  description="We couldn't reach the Prometheus server. Either the server no longer exists or the configuration details need updating."
  primarybuttonlink="/path/to/settings"
  primarybuttontext="Verify configuration"
  svgpath="/path/to/empty-group-illustration.svg"
  title="Connection failed"
/>
`;

exports[`GroupEmptyState Renders an empty state for CONNECTION_FAILED 2`] = `undefined`;

exports[`GroupEmptyState Renders an empty state for FOO STATE 1`] = `
<gl-empty-state-stub
  compact="true"
  description="An error occurred while loading the data. Please try again."
  svgpath="/path/to/empty-group-illustration.svg"
  title="An error has occurred"
/>
`;

exports[`GroupEmptyState Renders an empty state for FOO STATE 2`] = `undefined`;

exports[`GroupEmptyState Renders an empty state for LOADING 1`] = `
<gl-empty-state-stub
  compact="true"
  description="Creating graphs uses the data from the Prometheus server. If this takes a long time, ensure that data is available."
  svgpath="/path/to/empty-group-illustration.svg"
  title="Waiting for performance data"
/>
`;

exports[`GroupEmptyState Renders an empty state for LOADING 2`] = `undefined`;

exports[`GroupEmptyState Renders an empty state for NO_DATA 1`] = `
<gl-empty-state-stub
  compact="true"
  svgpath="/path/to/empty-group-illustration.svg"
  title="No data to display"
/>
`;

exports[`GroupEmptyState Renders an empty state for NO_DATA 2`] = `"The data source is connected, but there is no data to display. <a href=\\"/path/to/docs\\">More information</a>"`;

exports[`GroupEmptyState Renders an empty state for TIMEOUT 1`] = `
<gl-empty-state-stub
  compact="true"
  svgpath="/path/to/empty-group-illustration.svg"
  title="Connection timed out"
/>
`;

exports[`GroupEmptyState Renders an empty state for TIMEOUT 2`] = `"Charts can't be displayed as the request for data has timed out. <a href=\\"/path/to/docs\\">More information</a>"`;

exports[`GroupEmptyState Renders an empty state for UNKNOWN_ERROR 1`] = `
<gl-empty-state-stub
  compact="true"
  description="An error occurred while loading the data. Please try again."
  svgpath="/path/to/empty-group-illustration.svg"
  title="An error has occurred"
/>
`;

exports[`GroupEmptyState Renders an empty state for UNKNOWN_ERROR 2`] = `undefined`;
